packages/riipen-ui/src/components/TableGenerator.jsx

```diff
@@ -176,23 +193,34 @@ class TableGenerator extends React.Component {
       return this.renderEmpty();
     }
 
+    const isExpandable = expandRow && expandedNode;
+
     return (
       <React.Fragment>
         {this.renderTableHeader()}
         <TableBody>
           {data.map((row, i) => {
             return (
-              <TableRow hover={hover} key={`default-row-${i}`}>
-                {columns.map((col, j) => (
-                  <TableCell
-                    classes={[linkedStyles.className]}
-                    key={`default-cell-${i}-${j}`}
-                    align={this.getCellAlignment(col)}
-                  >
-                    {col?.cell(row, isMobile)}
-                  </TableCell>
-                ))}
-              </TableRow>
+              <React.Fragment key={`default-row-${i}`}>
+                <TableRow onClick={row.onClick} hover={hover}>
+                  {columns.map((col, j) => (
+                    <TableCell
+                      classes={[linkedStyles.className]}
+                      key={`default-cell-${i}-${j}`}
+                      align={this.getCellAlignment(col)}
+                    >
+                      {col?.cell(row, isMobile)}
+                    </TableCell>
+                  ))}
+                </TableRow>
+                {isExpandable && expandRow(row) && (
+                  <TableRow>
+                    <TableCell colSpan={columns.length}>
+                      {expandedNode(row)}
+                    </TableCell>
+                  </TableRow>
+                )}
```

Comment:
I dont have time to pull this locally at the moment so just something to look into for the morning, I think this wont work correctly with the hover prop. According to the designs the main row and hovered row should be highlighted when moused over. I dont know if this is possible as I dont know exactly how moving the cells into the parent row would work (would it wrap the row?) Something to look into for now

Reply:
yea the expanded stuff does not get "hovered" when you hover over the row. i guess for consistency i should add it